src/ui/actions/GetTasksToShowAction.cpp

```diff
@@ -4,24 +4,21 @@
 
 #include "GetTasksToShowAction.h"
 
+GetTasksToShowAction::GetTasksToShowAction(const std::optional<Core::TaskID> &id) : id_{id} {
+}
+
 GetTasksToShowAction::GetTasksToShowAction(const std::string &arg) : arg_{arg} {
 }
 
 ActionResult GetTasksToShowAction::execute(const std::shared_ptr<ModelInterface> &model) {
-    std::optional<Core::TaskID> id{Core::TaskID()};
-    try {
-        id->set_value(std::stoi(arg_));
-        if (!model->IsPresent(*id))
-            return {ActionResult::Status::ID_NOT_FOUND, id};
-    } catch (const std::invalid_argument &) {
-        id = std::nullopt;
-    }
+    if (id_ && !model->IsPresent(*id_))
+        return {ActionResult::Status::ID_NOT_FOUND, id_};
 
     std::vector<Core::TaskEntity> tasks;
-    if (arg_.empty())
+    if (!id_ && arg_.empty())
         tasks = model->getTasks();
-    else if (id)
-        tasks = model->getTaskWithSubtasks(*id);
+    else if (id_)
+        tasks = model->getTaskWithSubtasks(*id_);
```

Comment:
Why using if statement to define different behaviours? Consider creating different Actions, for showing by id, label or show all tasks. In this case, your GetTasksToShowActions changes it's behaviour depending on its argument (id or string - label).